docs/tutorial.md

```diff
@@ -0,0 +1,231 @@
+      resource("data/airplane"),
+      Csv,
+      SaveMode.ErrorIfExists,
+      options = Map("header" -> "true", "inferSchema" -> "true"))
+ }
+```
+Congratulations, you have built your first data lake. It has one environment: `test`, which contains one `DataLink`: a csv file loaded from your local disk. You cataloged that data to be in the `raw` zone of your data lake and it is named `airplane`. Let's extend this with two weather sources, in a similar fashion, and wrap it all in our very own `AirplaneDatalake` object.
+
+```scala
+object AirplaneDatalake extends Datalake {
+     environment("test") { refs =>
+        refs += DataUID("raw", "airplane") -> new FileSystemDataLink(
+          resource("data/airplane"),
+          Csv,
+          SaveMode.ErrorIfExists,
+          options = Map("header" -> "true", "inferSchema" -> "true"))
+          
+        refs += DataUID("raw.weather", "daily") -> new FileSystemDataLink(
+            resource("data/weather/daily"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map("header" -> "true", "inferSchema" -> "true"))
+        
+        refs += DataUID("raw.weather", "station") -> new FileSystemDataLink(
+            resource("data/weather/station"),
+            Csv,
+            SaveMode.ErrorIfExists,
+            options = Map(
+              "header" -> "true", 
+              "inferSchema" -> "true", 
+              "delimiter"   -> "|"))
+     }
+     
+     private def resource(path: String): String = File.resource(path).pathAsString
+ }
+``` 
+
+Great, now how can you use this? Well, from this moment onwards, you can always access the data sources by just asking the data lake:
+
+```scala
+    val df = AirplaneDatalake("raw" -> "airplane").read()
+```
+This allows you to define all the data sources in your data lake in one single location, and you can use them wherever you need them. You don't need to worry about which format it's in, the exact delimiter to set, ... That's all abstracted away from you. And while this does not add a lot of value if it's just you and a small team, doing a very specific analysis, this can be useful in medium to large organisations, where you have multiple teams of engineers working on the data lake, there is high reuse of data and data scientists just want quick access to all available data. 
+ 
+### Build a data pipeline
+Having a lot of data in your data lake is only valuable once you start doing things with it. This usually involves building some sort of pipeline. There's nothing wrong with simply calling a bunch of Spark transformations in a `run()` function, and simply executing it. And Lighthouse allows you to do exactly that. 
+
+However, in larger projects, this can quickly result in very long, hard to understand, hard to test, and hard to reuse pipelines. So how can we improve on this? By using the `Pipeline` library available in Lighthouse. Let's look at an example:
```

Comment:
However, in larger projects this (remove comma)
hard to test and (remove comma)
how can we improve this (remove on)